src/test/mocks/MockEngine.sol

```diff
@@ -16,7 +17,7 @@ import "../../config/errors.sol";
  * @title   MockEngine
  * @notice  Implement execute to test all flow in BaseEngine
  */
-contract MockEngine is BaseEngine {
+contract MockEngine is BaseEngine, ReentrancyGuard {
```

Comment:
I moved `ReentrancyGuard` from the BaseEngine to the inheriting contracts. I needed to replace it with the upgradable version in FMV2. Since it was removed from the base class it needed to be added here in order to keep `nonReentrant` on `execute`.

It's better to keep it so that we keep it as similar to a real engine as possible.